casadm/cas_lib.c

```diff
@@ -2157,6 +2161,74 @@ static inline const char *partition_get_csv_col(CSVFILE *csv, int col,
 	return val;
 }
 
+static int calculate_max_allocation(uint16_t cache_id, const char *allocation,
+		uint32_t *part_size)
+{
+	uint32_t value;
+
+	if (strnlen(allocation, MAX_STR_LEN) == 1) {
+		/* Case with '0' or '1' */
+		value = strtoul(allocation, NULL, 10);
+		if (value > 1)
+			return FAILURE;
+
+		*part_size = value * 100;
+
+		return SUCCESS;
+	} else if (strnlen(allocation, MAX_STR_LEN) == 4) {
```

Comment:
Maybe instead of making separate cases for all possible string lengths we could just parse this number as fixed point number with optional radix point and fractional part, so that we could support not only "1" or "1.00" but also "1.0" notion.

Reply:
simple sscanf_s(allocation, "%d.%d") would do the job, just make sure to check return value to know whether optional part after the dot was read

Reply:
actually I don't see why not  "sscanf_s("allocation, "%f"); "

Reply:
I used both `%s` and `%f` to make validation and parsing easier.